src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1529,6 +1530,24 @@ public function getContent($asResource = false)
         return $this->content;
     }
 
+    /**
+     * Return request body content as a json decoded array.
+     *
+     * @return mixed
+     *
+     * @throws JsonException
+     */
+    public function getJsonDecoded()
+    {
+        $json = json_decode($this->getContent(), true, 512, JSON_BIGINT_AS_STRING);
+
+        if (JSON_ERROR_NONE !== ($errorCode = json_last_error())) {
+            throw new JsonException('Unable to parse JSON: '.json_last_error_msg(), $errorCode);
```

Comment:
We usually have dots at the end of sentences in exception messages. I think, we should do it here too.

```suggestion
            throw new JsonException(sprintf('Unable to parse JSON: %s.', json_last_error_msg()), $errorCode);
```